src/front/wgsl/lexer.rs

```diff
@@ -215,6 +228,32 @@ impl<'a> Lexer<'a> {
         }
     }
 
+    pub(super) fn expect_generic(&mut self, expected: Token<'a>) -> Result<(), Error<'a>> {
+        let token = self.next_generic();
+        if token == expected {
+            Ok(())
+        } else {
+            let description = match expected {
```

Comment:
this whole match should be removed given that we know it's a paren

Reply:
Alternatively we could extract the description match into a separate function here to avoid the duplication. I've pushed that version first, let me know what you think. Otherwise I'll change it to your recommendation.